.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,266 @@
+name: "Continuous Integration"
+
+on:
+  pull_request:
+    paths-ignore:
+      - 'mkdocs.yml'
+      - '**.md'
+  push:
+    branches:
+      - '[0-9]+.[0-9]+.x'
+      - 'refs/pull/*'
+    paths-ignore:
+      - 'mkdocs.yml'
+      - '**.md'
+
+env:
+  COMPOSER_ARGS_DEFAULT: --ansi --no-interaction --no-progress --prefer-dist
+  COMPOSER_ARGS_PHP8: --ansi --no-interaction --no-progress --prefer-dist --ignore-platform-req=php
+  CONFIG_FILE: .github/workflows/continuous-integration-config.json
+  PHP_DEFAULT_EXTENSIONS: '["mbstring","intl","readline"]'
+  PHP_DEFAULT_INI: '["memory_limit=-1"]'
+
+jobs:
+  lint-json:
+    name: Lint JSON files
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: json-syntax-check
+        run: |
+          echo "Linting composer.json"
+          jq empty composer.json
+          for JSON in .github/workflows/*.json;do
+            echo "Linting $JSON"
+            jq empty $JSON
+          done
+
+  dependency-versions-matrix:
+    name: Dependencies to Test Matrix
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.dependency-versions-matrix.outputs.check }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: dependency-versions-matrix
+        run: |
+          DEPS=$(if [ -f composer.lock ]; then echo -n '["lowest", "locked", "latest"]'; else echo -n '["lowest", "latest"]'; fi)
+          echo "Testing against dependencies: ${DEPS}"
+          echo "::set-output name=check::${DEPS}"
+
+  config:
+    name: Retrieve configuration
+    runs-on: ubuntu-latest
+    needs: [lint-json]
+    outputs:
+      json: ${{ steps.config.outputs.json }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: config
+        run: |
+          CONFIG=$(if [ -f ${CONFIG_FILE} ]; then jq -c '.' < ${CONFIG_FILE} ; else echo -n "{}" ; fi)
+          echo "Using config: $CONFIG"
+          echo "::set-output name=json::${CONFIG}"
+
+  supported-versions-matrix:
+    name: Supported Versions Matrix
+    runs-on: ubuntu-latest
+    needs: [lint-json]
+    outputs:
+      version: ${{ steps.supported-versions-matrix.outputs.version }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: supported-versions-matrix
+        uses: WyriHaximus/github-action-composer-php-versions-in-range@v1
+
+  supported-checks-matrix:
+    name: Determine which checks to run
+    runs-on: ubuntu-latest
+    needs: [config]
+    outputs:
+      check: ${{ steps.supported-checks-matrix.outputs.check }}
+    env:
+      CONFIG_JSON: ${{ needs.config.outputs.json }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: supported-checks-matrix
```

Comment:
I think the entire "supported checks" should be moved from `laminas.dev` to a dedicated github action, so that it runs locally, and no HTTP traffic is done to decide what to run in CI here.

Reply:
I'm going to look into this, as I think it may be a more scalable approach; we can likely reduce 7 of the steps down to 1. Even if that one step is slower than any of the other individual steps, it will likely run faster over the run of the entire workflow.